### PR TITLE
[ESLint-plugin] Bump every eslint packages

### PR DESCRIPTION
**Pull request by alexandcote**
This PR bumped every ESlint packages to their last version. This new version will support the last version of typescript with type import 🎉

**I listed the most important changes.**

#### Changelogs
- [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/blob/master/CHANGELOG)
- [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier/blob/master/CHANGELOG.md)
- [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import/blob/master/CHANGELOG.md)
- [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest/blob/master/CHANGELOG.md)
- [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node/releases)
- [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md)
- [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-script-url.md)

#### New rules I enabled
-  [eslint-plugin-react/jsx-no-script-url](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-script-url.md)

#### New rules I'm not sure if we should enable
- [eslint-plugin-react/jsx-no-literals](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-literals.md)

#### New rules I don't thing we need to enable
- [typescript-eslint/method-signature-style](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/method-signature-style.md)
- [typescript-eslint/prefer-reduce-type-parameter](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-reduce-type-parameter.md)
- [typescript-eslint/prefer-ts-expect-error](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-ts-expect-error.md)
- [typescript-eslint/no-unsafe-assignment](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unsafe-assignment.md)
- [typescript-eslint/init-declarations](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/init-declarations.md)
- [typescript-eslint/keyword-spacing](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/keyword-spacing.md)
- [eslint-plugin-react/require-default-props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md)
- [eslint-plugin-react/function-component-definition](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md)
- [eslint-plugin-react/no-adjacent-inline-elements](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-adjacent-inline-elements.md)
- [eslint-plugin-react/forbid-component-props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-component-props.md)
- [eslint-plugin-react/forbid-dom-props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-dom-props.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,13 +21,12 @@
     "@shopify/tophat": "^0.0.4",
     "codecov": "^3.6.5",
     "eslint": "^6.8.0",
-    "eslint-plugin-prettier": "^3.1.2",
```
Comment (vsumner) [inline package.json:24]: Was this intentional?

File: packages/eslint-plugin/lib/config/rules/react.js [hunk 2]
```diff
@@ -99,6 +99,8 @@ module.exports = {
   'react/style-prop-object': 'error',
   // Prevent void DOM elements (e.g. <img />, <br />) from receiving children
   'react/void-dom-elements-no-children': 'error',
+  // Prevent usage of javascript: in URLs
```
Comment (vsumner) [inline packages/eslint-plugin/lib/config/rules/react.js:102]: Nice

File: packages/eslint-plugin/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@shopify/eslint-plugin",
-  "version": "36.1.0",
+  "version": "37.0.0-packages-bump-beta.1",
```
Comment (vsumner) [inline packages/eslint-plugin/package.json:3]: You'll need to reset this.

File: package.json (hunk 1 above)
Comment (alexandcote) [reply package.json:24]: No! Good catch

Comment (alexandcote) [reply package.json:24]: Ahhh yeah this is intentional, we use the shopify eslint plugin who contain this plugin.

Comment (vsumner) [reply package.json:24]: Nice deduping then 💯

File: packages/eslint-plugin/lib/config/rules/react.js (hunk 2 above)
Comment (TheMallen) [reply packages/eslint-plugin/lib/config/rules/react.js:102]: I wish there was a "this" reaction

**Pull request merged**